tests/pytests/test_aux_save2.py

```diff
@@ -0,0 +1,278 @@
+import subprocess
+from common import *
+from RLTest import Env
+
+RDBS = [
+    'redisearch_2.8.18_empty_dict.rdb',
+    'redisearch_2.10.7_empty_dict.rdb',
+]
+
+@skip(cluster=True, no_json=True, asan=True)
+def testLoadRdbWithoutIndexAuxData(env: Env):
+    env.expect('FT.CREATE', 'idx', 'SCHEMA', 't', 'TEXT').equal('OK')
+    env.expect('HSET', 'doc1', 't', 'hello').equal(1)
+    env.expect('FT.SEARCH', 'idx', 'hello', 'NOCONTENT').equal([1, 'doc1'])
+    env.expect('FT.DROPINDEX', 'idx').equal('OK')
+    # Save state to RDB
+    env.stop()
+    # Restart without modules
+    env.assertEqual(len(env.envRunner.modulePath), 2)
+    env.assertEqual(len(env.envRunner.moduleArgs), 2)
+    env.envRunner.modulePath.pop()
+    env.envRunner.moduleArgs.pop()
+    env.envRunner.modulePath.pop()
+    env.envRunner.moduleArgs.pop()
+    env.envRunner.masterCmdArgs = env.envRunner.createCmdArgs('master')
+    # Attempt to load RDB should work because the RDB
+    # does not contains module aux data
+    env.start()
+    env.expect('HGET', 'doc1', 't').equal('hello')
+
+@skip(cluster=True, no_json=True, asan=True)
+def testLoadRdbWithIndexAuxData(env: Env):
+    env.expect('FT.CREATE', 'idx', 'SCHEMA', 't', 'TEXT').equal('OK')
+    env.expect('HSET', 'doc1', 't', 'hello').equal(1)
+    env.expect('FT.SEARCH', 'idx', 'hello', 'NOCONTENT').equal([1, 'doc1'])
+    # Save state to RDB
+    env.stop()
+    # Restart without modules
+    env.assertEqual(len(env.envRunner.modulePath), 2)
+    env.assertEqual(len(env.envRunner.moduleArgs), 2)
+    env.envRunner.modulePath.pop()
+    env.envRunner.moduleArgs.pop()
+    env.envRunner.modulePath.pop()
+    env.envRunner.moduleArgs.pop()
+    env.envRunner.masterCmdArgs = env.envRunner.createCmdArgs('master')
```

Comment:
Consider moving to a helper function (reduce repetition)

Reply:
OK. Helper function created, but maybe we should add that functionality to RLTest because we are going to need it for CONFIG params tests also.